app/main.py

```diff
@@ -1,3 +1,29 @@
-def cache(func):
-    # Write your code here
-    pass
+import hashlib
+import json
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cache = {}
+
+    def compute_key(args: tuple, kwargs: dict) -> str:
+        key = hashlib.md5(
+            json.dumps(args, sort_keys=True).encode("utf-8")
+        ).hexdigest()
+        key += hashlib.md5(
+            json.dumps(kwargs, sort_keys=True).encode("utf-8")
+        ).hexdigest()
+        return key
+
+    def wrapper(*args, **kwargs) -> Any:
+        key = compute_key(args, kwargs)
+        if key in cache:
```

Comment:
try  **if args not in cache:**  instead  **key = compute_key(args, kwargs), if key in cache:**